File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    type = {'R':0,'T':0, 'C':0, 'F':0, 'J':0, 'M':0, 'A':0 , 'N':0}
    scores = [0,3,2,1,0,1,2,3]
    
    for (let i = 0 ; i < survey.length; i ++) {
        su = survey[i]
        su = su.split('') // 리스트로 변환, 비동의: 첫번째, 동의 : 두번째   
        let target;
        
        if (choices[i] >= 5) { // 동의 쪽
            target = su[1]
            type[target] += scores[choices[i]]
        }else {
            target = su[0];
            type[target] += scores[choices[i]]
        }
    }   
            
    const ty = Object.keys(type)
    
    for (let i = 0; i < ty.length; i+=2) {
        if (type[ty[i]] < type[ty[i+1]]) {
            answer += ty[i+1]
        } else {
            answer += ty[i]
        }
    }

    return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    p = {'R':0, 'T':0,'C':0,'F':0,'J':0,'M':0,'A':0,'N':0}
    
    for i in range(len(survey)):
        if choices[i] <= 3:
            p[survey[i][0]] += 4 - choices[i]
        
        else:
            p[survey[i][1]] += choices[i] - 4 
            
    if p['R'] >= p['T']:
        answer += 'R'
    if p['R'] < p['T']:
        answer += 'T'
    
    if p['C'] >= p['F']:
        answer += 'C'
    if p['C'] < p['F']:
        answer += 'F'

    if p['J'] >= p['M']:
        answer += 'J'
    if p['J'] < p['M']:
        answer += 'M'
    
    if p['A'] >= p['N']:
        answer += 'A'
    
    if p['A'] < p['N']: 
        answer += 'N'
     
    
        
    
    return answer